components/dashboard/src/contexts/FeatureFlagContext.tsx

```diff
@@ -55,31 +55,40 @@ const FeatureFlagContextProvider: React.FC = ({ children }) => {
                 personalAccessTokensEnabled: { defaultValue: false, setter: setPersonalAccessTokensEnabled },
                 slow_database: { defaultValue: false, setter: setUseSlowDatabase },
             };
+
             for (const [flagName, config] of Object.entries(featureFlags)) {
-                if (teams) {
-                    for (const team of teams) {
+                const value = () => {
+                    // First evaluate if the flag is enabled for the user
+                    const valueForUser = await getExperimentsClient().getValueAsync(flagName, config.defaultValue, {
+                        user,
+                        projectId: project?.id,
+                        teamId: team?.id,
+                        teamName: team?.name,
+                    });
+
+                    if (valueForUser) {
+                        return valueForUser;
+                    }
+
+                    // Check if the flag is enabled for any of the user's teams
+                    for (const team of teams || []) {
                         const flagValue = await getExperimentsClient().getValueAsync(flagName, config.defaultValue, {
                             user,
                             projectId: project?.id,
                             teamId: team.id,
                             teamName: team?.name,
                         });
 
-                        // We got an explicit override value from ConfigCat
-                        if (flagValue !== config.defaultValue) {
-                            config.setter(flagValue);
-                            return;
```

Comment:
Here, we'd return early. But this would actually stop the evaluation of other feature flags so they wouldn't get updated.